lib/ui/custom.dart

```diff
@@ -126,3 +223,31 @@ class SelectableList<T> extends CoreElement {
     _selectionController.add(item);
   }
 }
+
+class TreeToggle extends CoreElement {
+  TreeToggle({bool empty = false})
+      : super('div', classes: 'tree-toggle octicon') {
+    if (!empty) {
+      click(() {
+        _isOpen = !_isOpen;
+        _openController.add(_isOpen);
+        toggleClass('octicon-triangle-right', !_isOpen);
```

Comment:
these icons are probably better than the ones from material that I'm currently abusing.